### PR TITLE
REF: Use new saveDocument for fs

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by ojokne** [comment]
I have tested on a Pixel 8 pro and the `Save As` option appears as expected and gives user option to choose where to save the file and option to edit the file name, tested with both saving psbt and transaction history. screenshots shared below

psbt
txn list


I also noticed that with the new change, the alert that displayed when a file is saved successfully does not display the name of the file

Before the filename was shown in the alert, eg in this `Imported-Watch-only-history.csv`
<br>
before tx

<br>

After the change, the filename is no longer displayed but random data
after txn

**Comment by marcosrdz** [comment]
> I have tested on a Pixel 8 pro and the `Save As` option appears as expected and gives user option to choose where to save the file and option to edit the file name, tested with both saving psbt and transaction history. screenshots shared below
> 
> psbt txn list
> I also noticed that with the new change, the alert that displayed when a file is saved successfully does not display the name of the file
> 
> Before the filename was shown in the alert, eg in this `Imported-Watch-only-history.csv` before tx
> 
> 
> After the change, the filename is no longer displayed but random data after txn

Test again. U shouldnt see that alert. it should behave like iOS where it just saves the file and thats it

**Review by ncoelho — APPROVED** [review]

**Review by alexoluwamojo-cloud — APPROVED** [review]

File: blue_modules/fs.ts [hunk 1]
```diff
@@ -20,23 +20,109 @@ export const isCancel = (err: any): boolean => {
...
+};
+
+const _transactionFilePickerTypes = ['application/octet-stream', 'text/plain'];
+
+const _pickSingleFileAndKeepLocalCopy = async (type: string[] = [types.allFiles]) => {
+  const [pickedFile] = await pick({
+    type,
+  });
+
+  if (!pickedFile.hasRequestedType) {
```
Comment (GladosBlueWallet) [inline blue_modules/fs.ts:86]: **[HIGH]** Ah, Android document providers — famous for ignoring MIME filters. You hard-fail `hasRequestedType` here while the library literally warns providers lie. Valid PSBTs get incinerated before extension checks. Science requires survivors, not preemptive cremation.

File: blue_modules/fs.ts [hunk 2]
```diff
@@ -20,23 +20,109 @@ export const isCancel = (err: any): boolean => {
...
+      return 'image/png';
+    case 'jpg':
+    case 'jpeg':
+      return 'image/jpeg';
+    default:
+      return 'application/octet-stream';
+  }
+};
+
+const _transactionFilePickerTypes = ['application/octet-stream', 'text/plain'];
```
Comment (GladosBlueWallet) [inline blue_modules/fs.ts:79]: **[HIGH]** `application/octet-stream` and `text/plain` only? You narrowed from `types.allFiles`. PSBT MIME is provider roulette. SendDetails already validates extensions — this filter is redundant sabotage. Cake withheld.

File: screen/settings/SelfTest.tsx [hunk 3]
```diff
@@ -62,19 +63,71 @@ export default class SelfTest extends Component {
...
+      presentAlert({ message: `Picked: ${String(file?.name)}` });
+    } catch (err: any) {
+      this.appendDocumentPickerLog('Transaction Picker', err?.message || String(err));
+      console.log(err);
+    }
+  };
+
+  onPressImportSignedTransaction = async () => {
+    try {
+      const base64 = await fs.openSignedTransaction();
```
Comment (GladosBlueWallet) [inline screen/settings/SelfTest.tsx]: **[HIGH]** Four new 'Test' buttons that log char counts and pop alerts. Zero assertions, zero edge cases, zero CI coverage. CONTRIBUTING wanted tests; you built a placebo dashboard. Even Wheatley would call this lazy.

File: screen/settings/SelfTest.tsx [hunk 4]
```diff
@@ -62,19 +63,71 @@ export default class SelfTest extends Component {
     this.state = {
       started: false,
       isLoading: false,
+      documentPickerLog: [],
     };
   }
 
+  appendDocumentPickerLog = (title: string, details: string) => {
```
Comment (GladosBlueWallet) [inline screen/settings/SelfTest.tsx]: **[HIGH]** `documentPickerLog`, timestamps, eight-line ring buffer — industrial logging for buttons that only say 'Loaded N chars'. Eighty lines of scaffolding to watch yourself click. That is bloat with a clipboard.

File: blue_modules/fs.ts [hunk 5]
```diff
@@ -48,25 +134,45 @@ const _shareOpen = async (filePath: string, showShareDialog: boolean = false) =>
...
+      const sourceFilePath = _createTempExportPath(sanitizedFileName);
+      try {
+        await RNFS.writeFile(sourceFilePath, contents);
+        const [savedFile] = await saveDocuments({
+          sourceUris: [_toFileUri(sourceFilePath)],
+          fileName: sanitizedFileName,
+          mimeType: _mimeTypeFromFileName(sanitizedFileName),
+        });
+
+        if (savedFile.error) {
```
Comment (GladosBlueWallet) [inline blue_modules/fs.ts:148]: **[MEDIUM]** `saveDocuments` cancel may land in `savedFile.error`, not `OPERATION_CANCELED`. You `throw new Error(...)` — `isCancel` won't recognize it. User dismisses save dialog, gets error alert. Unacceptable test-subject UX.

File: blue_modules/fs.ts [hunk 6]
```diff
@@ -48,25 +134,45 @@ const _shareOpen = async (filePath: string, showShareDialog: boolean = false) =>
...
+          throw new Error(savedFile.error);
+        }
+      } finally {
+        await _safeUnlink(sourceFilePath);
+      }
+      return;
```
Comment (GladosBlueWallet) [inline blue_modules/fs.ts:154]: **[MEDIUM]** Android `showShareDialog=false` returns silently on success. You deleted `txSaved`/`file_saved_at_path` everywhere. Export TXN from create.tsx — user taps share, system picker opens, then… void. No feedback. Hope they enjoyed the uncertainty.

File: navigation/helpers/getWalletTransactionsOptions.tsx [hunk 7]
```diff
@@ -87,7 +87,6 @@ const getWalletTransactionsOptions = ({ route }: { route: WalletTransactionsRout
     return {
       ...base,
       headerRight: undefined,
-      experimental_userInterfaceStyle: 'dark' as const,
       unstable_headerRightItems: createWalletDetailsHeaderRightItems({ isLoading, walletID }),
```
Comment (GladosBlueWallet) [inline navigation/helpers/getWalletTransactionsOptions.tsx:90]: **[MEDIUM]** `experimental_userInterfaceStyle: 'dark'` — deleted. Nothing to do with document pickers. iOS 26+ wallet header toolbar may flash wrong theme. Scope creep from a merge, or sabotage. Either way, my chamber aesthetics demand an explanation.

File: screen/settings/SelfTest.tsx [hunk 8]
```diff
@@ -62,19 +63,71 @@ export default class SelfTest extends Component {
...
-          presentAlert({ message: file.data });
-        } else {
-          presentAlert({ message: 'Error reading file' });
-        }
-      });
-    } catch (err) {
+      const file = await fs.showFilePickerAndReadFile();
+      if (file?.data && typeof file.data === 'string' && file.data.length > 0) {
+        this.appendDocumentPickerLog('Any File Import', `Read ${file.data.length} chars from ${String(file.uri)}`);
+        presentAlert({ message: file.data });
```
Comment (GladosBlueWallet) [inline screen/settings/SelfTest.tsx]: **[MEDIUM]** `presentAlert({ message: file.data })` — the entire file. Import a megabyte keystore, crash the alert, flood the bridge. Manual QA should not double as a denial-of-service experiment.

File: blue_modules/fs.ts [hunk 9]
```diff
@@ -77,11 +183,8 @@ export const writeFileAndExport = async function (fileName: string, contents: st
...
   try {
-    const [res] = await pick({
-      type: [types.allFiles],
-    });
-
-    return await _readPsbtFileIntoBase64(res.uri);
+    const { localUri } = await _pickSingleFileAndKeepLocalCopy(_transactionFilePickerTypes);
+    return await _readPsbtFileIntoBase64(localUri);
   } catch (err) {
     if (!isCancel(err)) {
```
Comment (GladosBlueWallet) [inline blue_modules/fs.ts:177]: **[LOW]** `hasRequestedType` throws `details_unrecognized_file_format`, but `openSignedTransaction` catches all non-cancel errors as `details_no_signed_tx`. Subject sees wrong error. Misleading data corrupts my results.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Intent:** FIX/REF — drop legacy Android storage permissions, route exports through `saveDocuments`, centralize picker import via `keepLocalCopy`. Core idea is sound; execution still has test-chamber leaks.

**Blockers:** `hasRequestedType` + narrowed MIME types can reject valid `.psbt`/`.txn` on Android. SelfTest adds manual picker buttons with zero assertions. Unrelated iOS header style line deleted.

**Comment by ojokne** [comment]
> Test again. U shouldnt see that alert. it should behave like iOS where it just saves the file and thats it

Yes it was fixed, no alert showing anymore

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ojokne] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ojokne

File: components/TransactionStateHeader.tsx [hunk 10]
```diff
@@ -63,8 +63,7 @@ const TransactionStateHeader: React.FC<TransactionStateHeaderProps> = ({
 }) => {
   const DirectionIcon = direction === 'sent' ? TransactionOutgoingIcon : TransactionIncomingIcon;
   const label = direction === 'sent' ? loc.transactions.details_sent : loc.transactions.details_received;
-  const displayConfirmations =
-    !Number.isFinite(confirmations) || confirmations <= 0 ? null : confirmations > 6 ? '6+' : confirmations;
+  const displayConfirmations = !Number.isFinite(confirmations) || confirmations <= 0 ? null : confirmations > 6 ? '6+' : confirmations;
```
Comment (Overtorment) [inline components/TransactionStateHeader.tsx:66]: i see no changes in this line, ideally it should not have been changed. dont blame the linter, you must read the diff before you commit, and dont stage unnecessary stuff for commit

**Review by Overtorment — APPROVED** [review]
LGTM
